### PR TITLE
Replace LZCNT with BSR instruction in hdr_histogram

### PR DESCRIPTION
**Pull request by SeverinLeonhardt**
The LZCNT instruction produced by __lzcnt on Windows is not available
on many CPUs in use today. BSR is a safe replacement requiring only
minimal additional work.

### CODE REVIEW

**Comment by mpenick** [comment]
@SeverinLeonhardt  Thanks for researching and getting a pull request together. I just had one issue compiling on Windows, otherwise, LGTM.

File: src/third_party/hdr_histogram/hdr_histogram.cpp [hunk 1]
```diff
@@ -23,21 +23,36 @@
...
+  char isNonzero = _BitScanReverse64(&index, x);
+  if (isNonzero)
+    return index + 1;
+  else
+    return 0;
 #  else
   // On 32-bit this needs to be split into two operations
-  int32_t lz = (int32_t)__lzcnt((unsigned long)(x >> 32));
-  if (lz == 32) {
+  const char isNonzero = _BitScanReverse(&index, (unsigned long)(x >> 32));
```
Comment (mpenick) [inline src/third_party/hdr_histogram/hdr_histogram.cpp]: Should this be "const"?

**Comment by mpenick** [comment]
I noticed another issue too that's unrelated to the changes you made. For GCC-based compilers `__builtin_clzll(0)` is undefined so the code needs to check that value `x` for `0` (and `return 0`).

**Comment by SeverinLeonhardt** [comment]
Updated the pull request as you've suggested.

**Comment by mpenick** [comment]
Thanks for updating. Looks good.

**Pull request merged**